File: Rust/Week1/Day4/day4/src/bin/ex4.rs
use rand::Rng;
//NOTE: I added the dependency rand = "0.8" to cargo toml after some research.

fn main() {
    const WIDTH: usize = 10;
    const HEIGHT: usize = 10;
    let mut grid:[[usize;WIDTH];HEIGHT]= [[0; 10]; 10];
    let boss_threat = 50;
    let tile_with: [(&str, i32); 10] = [
        ("Nothing", 0), 
        ("Slime", 1),
        ("Goblin",3),
        ("Bandit",4),
        ("Wolf",5),
        ("Weak Orc",7),
        ("Orc",8),
        ("Dark Wizard",12),
        ("Troll",15),
        ("Dragon",boss_threat)
    ];

    tile_with[0].0;

    for x in 0..WIDTH {
        for y in 0..HEIGHT {
            grid[x][y] = rand::thread_rng().gen_range(0..10);
        };
    };

    let mut threat_count = 0;
    println!("Scanning area...");
    'scan: for (l_index, line) in grid.iter().enumerate() {
        for (c_index, cell) in line.iter().enumerate() {
            print!("Scaning cell ({l_index},{c_index})...");
            let present = tile_with[*cell].0;
            let threat = tile_with[*cell].1;
            //Note: I did the alocations above because string formatting doesn;t seem to like the tipple indexing as mentioned in ex 3
            threat_count += threat;
            if threat == boss_threat {
                println!("Boss found! It is a {present}! Its threat is {threat}! Ending search!");
                break 'scan;
            }
            println!("Therre is a {present}. Its threat is {threat}.");
        }
    }
    println!("Total threat count: {threat_count}");
}

File: CSharp/Week1/Day6/Day6/ex4.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Day6
{
    internal class ex4
    {
        const int WIDTH = 10;
        const int HEIGHT = 10;
        struct Cell
        {
            public String name;
            public int threat;
            public void redef(int threat)
            {
                this.threat = threat;
                this.name = this.threat switch
                {
                    0 => "Nothing",
                    1 => "a Goblin",
                    2 => "a Wolf",
                    3 => "an Orc",
                    4 => "a Troll",
                    _ => "a Dragon"
                };
            }
        };
        static void Main()
        {
            Cell[,] grid = new Cell[WIDTH, HEIGHT];
            Random rand = new Random();
            for(int x = 0; x < WIDTH; x++)
            {
                for(int y = 0; y < HEIGHT; y++)
                {
                    grid[x, y].redef(rand.Next(6));
                }
            }
            for (int x = 0; x < WIDTH; x++)
            {
                for (int y = 0; y < HEIGHT; y++)
                {
                    Console.WriteLine("Scaning ({0},{1})... You see {2}.", x, y, grid[x, y].name);
                    if (grid[x,y].threat == 5)
                    {
                        Console.WriteLine("A boss was found! Ending search!");
                        goto EndLoops;
                    }
                }
            }
        EndLoops:
            return;
        }
    }
}
